be/tool/beck.c

```diff
@@ -1566,6 +1595,168 @@ static void builder_fini(struct builder *b)
 	       b->b_act, b->b_tx);
 }
 
+static void override_be_cfg_defaults(struct m0_be_domain_cfg *cfg,
+			     	     const char              *str_key,
+			     	     const char              *str_value)
+{
+	uint64_t  value1_64;
+	uint64_t  value2_64;
+	char     *s1;
+	char     *s2;
+	bool      value_overridden = true;
+
+	if (m0_streq(str_key,"bec_tx_active_max"))
+		cfg->bc_engine.bec_tx_active_max = m0_strtou64(str_value, 0, 10);
+	else if (m0_streq(str_key, "bec_group_nr"))
+		cfg->bc_engine.bec_group_nr = m0_strtou64(str_value, 0, 10);
+	else if (m0_streq(str_key, "tgc_tx_nr_max"))
+		cfg->bc_engine.bec_group_cfg.tgc_tx_nr_max = m0_strtou64(str_value,
+									 0, 10);
+	else if (m0_streq(str_key, "tgc_size_max")) {
+		s1 = m0_strdup(str_value);
```

Comment:
All the code of this function assumes the key and value have a correct format.
Otherwise, this function may not be able to handle them correctly.
E.g. The str_value has a valid decimal number, 
the str_value has a "dec_number,dec_number" in this case.

We can assume this, because this YAML file is prepared by Admin.
Otherwise, strict error error handling must be performed.

One possible checking needed is:
checking the result of m0_strtou64(str_value, 0, 10).
Also, retrieve the second value from the val_string.
if it is needed and the returned value is valid, it is OK.
Otherwise, report error.

Reply:
OK